data-prepper-core/src/main/java/org/opensearch/dataprepper/DataPrepper.java

```diff
@@ -71,6 +75,28 @@ public DataPrepper(
      */
     public boolean execute() {
         peerForwarderServer.start();
+        Set<String> pipelineNames = transformationPipelines.keySet();
+        int numRetries = 0;
+        while (pipelineNames.size() > 0 && numRetries++ < MAX_RETRIES) {
+            Set<String> failedPipelines = new HashSet<String>();
```

Comment:
`failedPipelines` seems a little inaccurate. Perhaps `uninitializedPipelines`? At first glance it seems that these are failed for good.